src/hotspot/cpu/riscv/frame_riscv.inline.hpp

```diff
@@ -186,7 +186,7 @@ inline bool frame::equal(frame other) const {
               unextended_sp() == other.unextended_sp() &&
               fp() == other.fp() &&
               pc() == other.pc();
-  assert(!ret || ret && cb() == other.cb() && _deopt_state == other._deopt_state, "inconsistent construction");
+  assert(!ret || (cb() == other.cb() && _deopt_state == other._deopt_state), "inconsistent construction");
```

Comment:
Per comments in reviews for similar changes for other ports, I've removed the unnecessary "ret &&".
Builds fine, but should probably be (re)tested by a port maintainer.